python/lsst/ts/mtdome/mock_llc/amcs.py

```diff
@@ -259,6 +283,11 @@ async def fans(self, action: str) -> float:
         action: `str`
             The value should be ON or OFF but the value doesn't get validated
             here.
+
+        Returns
+        -------
+        `float`
+            The expected duration of the command.
```

Comment:
I strongly suggest either returning a duration (or end timestamp, which is apparently what you used to do?) *or* saving the estimated end time, but not both.

I am very uncomfortable with saving a duration (as self.duration) because it needs an associated timestamp in order to be useful. And if you need a second attribute value to interpret duration then you have to keep two different attributes in sync. It is much safer and simpler to just save the end time.

Also doing something two different ways (both returning a value and saving it as an attribute) is confusing to the reader makes code maintenance more difficult, since some code may rely on the return value while other code relies on the saved attribute.

Reply:
The method has always returned a duration. The only thing that changed here was to add a Return section in the doc string. The durations are checked in the unit tests which is why they are both stored internally and returned. I prefer to keep the code this way.

Reply:
It really makes me nervous to store durations instead of timestamps. But since it is mock code instead of main library code I won't insist.